### ISSUE TITLE
Unsupported GPU architecture 'compute_60'

### ISSUE THREAD

**Issue by cgartists** [KlausT/ccminer-cryptonight#22]
Figures, I finally get a miner that seems promissing and it gives me this error. Does this not support the newest Nvidia Titan X Pascal cards?

Is there a way to make this mining software work with the Pascal architecture?

Commands I've entered, "./autogen  ./configure  make install"

make install gives the following error.

Making install in compat
make[1]: Entering directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[2]: Entering directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[3]: Entering directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[3]: Nothing to be done for 'install-exec-am'.
make[3]: Nothing to be done for 'install-data-am'.
make[3]: Leaving directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[2]: Leaving directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[1]: Leaving directory '/home/titan/Downloads/ccminer-cryptonight-master/compat'
make[1]: Entering directory '/home/titan/Downloads/ccminer-cryptonight-master'
gcc -DHAVE_CONFIG_H -I.  -I/usr/local/cuda/include -msse2  -fopenmp -pthread -fno-strict-aliasing  -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME   -g -O2 -MT ccminer-cpu-miner.o -MD -MP -MF .deps/ccminer-cpu-miner.Tpo -c -o ccminer-cpu-miner.o `test -f 'cpu-miner.c' || echo './'`cpu-miner.c
mv -f .deps/ccminer-cpu-miner.Tpo .deps/ccminer-cpu-miner.Po
gcc -DHAVE_CONFIG_H -I.  -I/usr/local/cuda/include -msse2  -fopenmp -pthread -fno-strict-aliasing  -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME   -g -O2 -MT ccminer-util.o -MD -MP -MF .deps/ccminer-util.Tpo -c -o ccminer-util.o `test -f 'util.c' || echo './'`util.c
mv -f .deps/ccminer-util.Tpo .deps/ccminer-util.Po
gcc -DHAVE_CONFIG_H -I.  -I/usr/local/cuda/include -msse2  -fopenmp -pthread -fno-strict-aliasing  -DSCRYPT_KECCAK512 -DSCRYPT_CHACHA -DSCRYPT_CHOOSE_COMPILETIME   -g -O2 -MT ccminer-sha2.o -MD -MP -MF .deps/ccminer-sha2.Tpo -c -o ccminer-sha2.o `test -f 'sha2.c' || echo './'`sha2.c
mv -f .deps/ccminer-sha2.Tpo .deps/ccminer-sha2.Po
nvcc -g -O2 -I . -D_FORCE_INLINES -I/usr/local/cuda/include -gencode=arch=compute_20,code=sm_20 -gencode=arch=compute_20,code=sm_21 -gencode=arch=compute_30,code=sm_30 -gencode=arch=compute_35,code=sm_35 -gencode=arch=compute_37,code=sm_37 -gencode=arch=compute_50,code=sm_50  -gencode=arch=compute_52,code=sm_52 -gencode=arch=compute_60,code=sm_60 -gencode=arch=compute_61,code=sm_61 --maxrregcount=128 --ptxas-options=-v  -o cryptonight/cryptonight.o -c cryptonight/cryptonight.cu
nvcc fatal   : Unsupported gpu architecture 'compute_60'
Makefile:1152: recipe for target 'cryptonight/cryptonight.o' failed
make[1]: *** [cryptonight/cryptonight.o] Error 1
make[1]: Leaving directory '/home/titan/Downloads/ccminer-cryptonight-master'
Makefile:726: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1

Looking forward to any help, thanks in advance!

**Comment by KlausT** [comment]
CUDA 7.5 doesn't support compute_60 and above.
If you don't want to install CUDA 8.0 then you could modify the makefile.am and delete the compute_60 and 61 stuff.

**Comment by cgartists** [comment]
Thanks! It now recognizes my GPU, however a new issue has come up. If I try to mine to a pool or even just benchmark my system it gives me this error.

ccminer --benchmark

-----------------------------------------------------------------
[2017-03-13 16:16:41] 1 miner threads started, using 'cryptonight' algorithm.
[2017-03-13 16:16:41] GPU #0: TITAN X (Pascal) (28 SMX), using 112 blocks of 8 threads

**GPU 0: invalid device function
cryptonight/cuda_cryptonight_extra.cu line 198**

In the file "cryptonight/cuda_cryptonight_extra.cu" line 198 says:

**exit_if_cudaerror(thr_id, __FILE__, __LINE__);**


NOTE: If I delete this or comment it out, it still gives me error line 198. What I find odd, if I remove some spacing from the document as a whole, exit_if_cudaerror(thr_id, __FILE__, __LINE__); should now be on different line. But when I re-run ccminer --background, It still says line 198 which now that line would read something different entirely.

Thanks again, very much appreciated!

**Comment by KlausT** [comment]
I guess the compiler didn't compile the modified file.
Try "make clean" before compiling it
But I don't think commenting out the error check will help.
Did you delete the compute_60 stuff? Try adding -gencode=arch=compute_52,code=sm_52

**Comment by cgartists** [comment]
In both files "Make" and "Makefile.am" I've removed the compute_60 stuff. 

**NVCC_GENCODE		= -D_FORCE_INLINES $(CUDA_CFLAGS) -gencode=arch=compute_20,code=sm_20 -gencode=arch=compute_20,code=sm_21 -gencode=arch=compute_30,code=sm_30 -gencode=arch=compute_35,code=sm_35 -gencode=arch=compute_37,code=sm_37 -gencode=arch=compute_50,code=sm_50  -gencode=arch=compute_52,code=sm_52**

Are there any other files that contain "60" architecture?  I when through the steps again,  using ./configure then removed the compute 60s. Then "make" and "make install" (which requires sudo to edit '/usr/local/bin/ccminer'). Through each step I checked and removed the "60s". It still gives the same error:

**GPU 0: invalid device function
cryptonight/cuda_cryptonight_extra.cu line 198**

Thanks again, I appreciate your help so far

**Comment by KlausT** [comment]
Oops, sorry, small typo in my comment.
I meant adding -gencode=arch=compute_52,code=compute_52 to the makefile.am
If this works then the first start of ccminer will probably take a little bit longer, but then work normally

**Comment by cgartists** [comment]
YES! It works now! Thanks! My hashrate is a little disappointing though. 1.6KH/s  I was in the megahash on ethereum, which is now unprofitable. Is there a way to increase the speed with some parameters?

Thanks very much for helping me make this work, appreciate your time.

**Comment by KlausT** [comment]
Try `-l 8x224 --bfactor=0` if you have enough memory

**Comment by cgartists** [comment]
Sweet! It's now at 2KH/s. I checked my Nvidia settings. Only 4GB vRAM being used out of 12GB. What parameters would I need to enter to utalize say 10GB?

**Comment by KlausT** [comment]
In this line:
`*** ccminer-cryptonight 2.01 (64 bit) for nVidia GPUs by tsiv and KlausT`
does it say 64bit, or does it say 32bit?

**Comment by cgartists** [comment]
64bit. My computer is also 64bit

**Comment by KlausT** [comment]
I made a small change. Maybe you could try 8x336 now, I can't say if it will work for you or not.

**Comment by cgartists** [comment]
8x336 still doesn't work, but my 8x224 hashrate is now 2.2KH/s. 200H faster.
